src/aiida/transports/plugins/ssh_auto.py

```diff
@@ -0,0 +1,50 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Plugin for transport over SSH (and SFTP for file transfer)."""
+
+import pathlib
+
+import paramiko
+
+from .ssh import SshTransport
+
+__all__ = ('SshAutoTransport',)
+
+
+class SshAutoTransport(SshTransport):
+    """Support connection, command execution and data transfer to remote computers via SSH+SFTP."""
+
+    _valid_connect_params = []
+    _valid_auth_options = []
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs, key_policy='AutoAddPolicy')
+
+        config_ssh = paramiko.SSHConfig()
+
+        with (pathlib.Path('~').expanduser() / '.ssh' / 'config').open() as handle:
+            config_ssh.parse(handle)
+
+        if self._machine not in config_ssh.get_hostnames():
```

Comment:
Is there any case where the connection would still work when the host is not in the `~/.ssh/config`? Is there a reason to not outright `raise` here?

Reply:
One consideration might be that the user might focus more on the (somewhat unclear) error and miss the very clear warning.

Reply:
Fair enough. One more reason why I decided to use a warning is that the `~/.ssh/config` is not the only place that SSH connections can be configured. There are also system wide configs like `etc/sshd/*` and others. These are also checked by `fabric` which is why it failed silently if the specified hostname is not part of the `~/.ssh/config` file, because it cannot conclude that there is a problem if that is the case. _We_ are adding this because for our purposes in most cases the user's config file would be the main place and we want to warn explicitly if we don't find the host there, but it is not necessarily a problem. Anyway, if others think it is better to make this an exception for now, happy to change it. We can always remove it later and turn it into a warning if there are cases where the plugin should rely on other configuration files as well.

Reply:
It's also not critical for me, and I see your point. In fact, now that I think of it, I might already know of a user that had a setup I wasn't familiar with, which might work fine even though the remote isn't specified in the `~/.ssh/config`. So let's keep the warning.

Reply:
@mbercx did you already tried the `core.ssh_auto` with your various real-life test cases? Did it work?

Reply:
Ah, [yes]! I tried 3 different clusters, with and without proxy jump. Also messed with the configuration to make it fail and then fixed it. All seems to work as expected.

So I think once we remove `ssh_fabric` we can go ahead and merge this, let it into the wild and see what happens. 😁